src/test/java/ro/unibuc/tbd/it/ClientServiceIT.java

```diff
@@ -0,0 +1,163 @@
+        meal.setName("Paste Carbonara");
+        meal.setIngredients(List.of("Paste", "Bacon"));
+        meal.setPrice((float) 27.99);
+        meal.setPortionSize(400);
+    }
+
+    @AfterEach
+    void cleanUp() {
+        clientRepository.deleteAll();
+        mealRepository.deleteAll();
+    }
+
+    @Test
+    void addToCart() {
+        // Arrange
+        when(clientRepository.findById(any())).thenReturn(Optional.ofNullable(client));
+        when(mealRepository.findById(any())).thenReturn(Optional.ofNullable(meal));
+        clientService.createClient(client);
+        mealService.createMeal(meal);
+
+        CartRequestDTO cartRequest = new CartRequestDTO();
+        cartRequest.setMealId(meal.getId());
+
+        // Act
+        Client result = clientService.addToCart(client.getId(), cartRequest);
+
+        // Assert
+        verify(clientRepository, times(2)).save(any());
+        assertTrue(result.getCart().containsKey(meal.getId()));
+    }
+
+    @Test
+    void addToCart_MealNotFound() {
+        // Arrange
+        when(clientRepository.findById(any())).thenReturn(Optional.ofNullable(client));
+        when(mealRepository.findById(any())).thenReturn(Optional.empty());
+        clientService.createClient(client);
+
+        CartRequestDTO cartRequest = new CartRequestDTO();
+        cartRequest.setMealId(meal.getId());
+
+        // Act + Assert
+        assertThrows(ResponseStatusException.class,
+                () -> clientService.addToCart(client.getId(), cartRequest),
+                "Expected addToCart() to throw a ResponseStatusException, but it didn't.");
+    }
+
+    @Test
+    void removeFromCart() {
+        // Arrange
+        when(clientRepository.findById(any())).thenReturn(Optional.ofNullable(client));
+        when(mealRepository.findById(any())).thenReturn(Optional.ofNullable(meal));
+        clientService.createClient(client);
+        mealService.createMeal(meal);
+
+        CartRequestDTO cartRequest = new CartRequestDTO();
+        cartRequest.setMealId(meal.getId());
+
+        // Act
+        clientService.addToCart(client.getId(), cartRequest);
+        Client result = clientService.removeFromCart(client.getId(), cartRequest);
+
+        // Assert
+        assertFalse(result.getCart().containsKey(meal.getId()));
+    }
+
+    @Test
+    void clearCart() {
+        // Arrange
+        when(clientRepository.findById(any())).thenReturn(Optional.ofNullable(client));
+        when(mealRepository.findById(any())).thenReturn(Optional.ofNullable(meal));
+
+        CartRequestDTO cartRequest = new CartRequestDTO();
+        cartRequest.setMealId(meal.getId());
+
+        // Act
+        clientService.addToCart(client.getId(), cartRequest);
+        Client result = clientService.clearCart(client.getId());
+
+        // Assert
+        verify(clientRepository, times(2)).save(any());
+        assertTrue(result.getCart().isEmpty());
+    }
+
+    @Test
+    void placeOrder() {
```

Comment:
we should later add a test checking that order placement throws exceptions, such as placeOrder_throwsWhenMealNotFound